opm/autodiff/LinearSolverAmgclNew.hpp

```diff
@@ -0,0 +1,309 @@
+/*
```

Comment:
I agree with @totto82 that the AmgCl stuff should be a separate PR. That notwithstanding, why are there two implementations, i.e.,  `LinearSolverAmgcl.hpp` and `LinearSolverAmgclNew.hpp`? (note: the term `new` should be avoided for code identifiers because what's new today is not new in 20 years anymore. the NAPI network code of the linux kernel illustrates this pretty well.)

Reply:
We will fix this.